code-challenges.test.js

```diff
@@ -16,35 +16,97 @@
 // a) Create a test with an expect statement using the variable provided. 
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
+describe("shuffleArray", () => {
+  it("takes in an array, removes the first item from the array and shuffles the remaining content.", () => {
+    expect(shuffleArray(colors1)).toEqual(expect.arrayContaining["yellow", "blue", "pink", "green"])
+    expect(shuffleArray(colors2)).toEqual(expect.arrayContaining["saffron", "aquamarine", "periwinkle", "indigo", "ochre"])
+  })
+})
+
+// Good Fail!
+// ReferenceError: shuffleArray is not defined
+
 const colors1 = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
 const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
-
 // b) Create the function that makes the test pass.
 
+// PSUEDO CODE:
+// Create a function named shuffleArray
+// .shift(1,5) to remove the first item from the array
+// I need to figure out how to shuffle the array after I slice purple from 
+// console log to check if it's working through node 
+// yarn jest to test my pass or fail
+
+const shuffleArray = () => {
+  let shuffledArray = colors1.sort(() => Math.random() - 0.5)
+  return colors1.slice(1, 5)
+}
+console.log(shuffleArray(colors1))
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the net total of votes.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = {upVotes: 13, downVotes: 2}
+describe("votes", () => {
+  it("takes in an object that contains up votes and down votes and returns the net total of votes.", () => {
+    expect(votes(votes1)).toEqual(11)
+    expect(votes(votes2)).toEqual(-31)
+  })
+})
+
+// Good Fail!
+// ReferenceError: votes is not defined
+
+const votes1 = { upVotes: 13, downVotes: 2 }
 // Expected output: 11
-const votes2 = {upVotes: 2, downVotes: 33}
+const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
 
-
 // b) Create the function that makes the test pass.
 
+// PSUEDO CODE:
+// Create a function named votes
+// I believe I should create a conditional statement that counts the votes and then totals them together
+// I would need a variable that starts a counter at 0
+// console log to check work
+// run yarn jest to test my pass or fail
+
+const votes = () => {
+  let counter = 0
+}
```

Comment:
I appreciate your effort here, Zack.  I think you're overthinking it.  It's basically asking given an object with 2 k-v pairs where the values are numbers, what is the net total of those numbers.  It all comes down to dot.notation.  I'm happy to work through this problem with you at some point.